packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -102,11 +102,15 @@ export type TreeObjectNode<
  *
  * @system @public
  */
-export type FieldHasDefault<T extends ImplicitFieldSchema> = [T] extends [
-	FieldSchema<FieldKind.Optional | FieldKind.Identifier>,
-]
-	? true
-	: false;
+export type FieldHasDefault<T extends ImplicitFieldSchema> =
+	// If the value is a node, then it is implicitly required, and therefore does not have a default.
```

Comment:
Oh, ya I missed that case!

Reply:
Actually, this case should be unnecessary, since the check below should return false for this case.

I suppose maybe it is this needed since ImplicitAllowedTypes is provided because TSC can't prove that no type extends both ImplicitAllowedTypes and FieldSchema?

Reply:
Actually, even in that case, it should get false. Was there a case that gave true? If so which one?

Reply:
Hmm. My intellisense must have been out-of-date or something. Reverting the change no longer fails the new tests 🤷‍♂️ Reverted.